.github/actions/setup-test-cluster/action.yaml

```diff
@@ -59,27 +59,28 @@ runs:
   using: composite
   steps:
     - name: Set up Go
-      uses: actions/setup-go@v4
+      uses: actions/setup-go@v5
       with:
         go-version: ${{ inputs.go-version }}
+        cache-dependency-path: api/go.sum
```

Comment:
We can now use the `cache-dependency-path` input to [indicate](https://github.com/actions/setup-go?tab=readme-ov-file#caching-dependency-files-and-build-outputs) files that should be cached instead of having to manually caching them using the `actions/actions/cache` action. The `actions/cache` action has also been removed since it's no longer needed. This applies for all the updated `actions/setup-[go/node/python]` actions.

This is similar done in the MLP PR https://github.com/caraml-dev/mlp/pull/100/files.